mentii/mentii
# Forgot password

This PR allows a user to reset their password through a series of steps.

**Significant Changes**
1. New email template
2. New attribute being added to a user item in the DB (only for the duration between forgot password and reset password)
3. Removal of test auth pages from the very beginning of the project
4. Some cleanup on sign in and register

**How to Test**
This is easiest to test using two different browsers at the same time.
1. Open Browser1 (B1) and Browser2 (B2)
2. Create a new account using B1
3. Activate account
4. Sign in with credentials in B1
5. Use B2 to click forgot password on the sign in page
6. Enter email used during registration
7. Open email and copy link to reset password
8. Open this link in B2
9. Reset the password for that user
10. Try to log in with the new password using B2 (should work)
11. Try to click around in B1 and you should be kicked out and sent back to login (since that auth token is no longer valid).

**Other Things to Test**
1. Try changing the password of a different user.
2. Try using the forgot password link more than once. The link itself will work but the reset password functionality will not.

**Notes**
😸
